File: miscellaneous/ltcd_817.py
'''
LeetCode - problem 817
'''

    def numComponents(self, head: ListNode, G: List[int]) -> int:
        count = 0
        i = head
        while i:
            if i.val in G:
                count += 1
                while i and i.val in G:
                    i = i.next
            else :
		i = i.next
        return count

File: miscellaneous/ltcd_817.cpp
//LeetCode - problem 817

    int numComponents(ListNode* head, vector<int>& G) {
        unordered_set<int> s(G.begin(), G.end());
        bool found = false;
        int n = 0;
        while (head != nullptr) {
            if (s.count(head->val)) {
                if (!found) {
                    found = true;  
                    n++;
                } 
            } else {
                found = false;
            }
            head = head->next;
        }
        return n;
    }
